### PR TITLE
HPCC-12789 Evaluate child queries after skips if not dependant

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@jamienoss please review.
Curiously this also improves the generated code for a few of the examples.  I think it is because aliases/rows are calculated which are then passed into the child queries, reducing the work they need to do.

File: ecl/hqlcpp/hqlhtcpp.cpp [hunk 1]
```diff
@@ -705,7 +705,7 @@ class DelayedStatementExecutor
                 if (worthHoisting)
                 {
                     //MORE: Remove || true to evaluate the subquery at the latest time.
```
Comment (jamienoss) [inline ecl/hqlcpp/hqlhtcpp.cpp]: This comment needs updating/removing.

**Comment by ghalliday** [comment]
@richardkchapman repushed after removing out of date comment.

**Comment by richardkchapman** [comment]
Cherry-picked into 5.6.0

**Pull request merged**